third_party/terraform/resources/resource_compute_region_instance_group_manager.go

```diff
@@ -714,21 +725,24 @@ func flattenRegionUpdatePolicy(updatePolicy *computeBeta.InstanceGroupManagerUpd
 	return results
 }
 
-func expandDistributionPolicy(configured *schema.Set) *computeBeta.DistributionPolicy {
-	if configured.Len() == 0 {
+func expandDistributionPolicy(d *schema.ResourceData) *computeBeta.DistributionPolicy {
+	dpz := d.Get("distribution_policy_zones").(*schema.Set)
+	dpts := d.Get("distribution_policy_target_shape").(string)
+	if dpz.Len() == 0 && dpts == "" {
```

Comment:
Is it possible to set the target shape if the distribution policy zones are empty? We are opening the possibility of an empty list of zones here, which seems like it might cause problems

Reply:
I don't think so. Also, there a number of limitations that appear once you set target shape which are not validated in terraform.

https://cloud.google.com/compute/docs/instance-groups/distributing-instances-with-regional-instance-groups#setting_a_policy_for_distributing_instances_across_zones